alicloud/resource_alicloud_cms_sitemonitor.go

```diff
@@ -0,0 +1,192 @@
+		return cmsClient.CreateSiteMonitor(request)
+	})
+	if err != nil {
+		return WrapError(err)
+	}
+
+	d.SetId(taskName)
+	return resourceAlicloudCmsSiteMonitorRead(d, meta)
+}
+
+func resourceAlicloudCmsSiteMonitorRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	cmsService := CmsService{client}
+
+	keyword := d.Get("task_name").(string)
+	siteMonitors, err := cmsService.DescribeSiteMonitor(keyword)
+	if err != nil {
+		return WrapError(err)
+	}
+
+	if len(siteMonitors) == 0 {
+		d.SetId("")
+		return nil
+	}
+	siteMonitor := siteMonitors[0]
+
+	d.SetId(siteMonitor.TaskId)
+	d.Set("address", siteMonitor.Address)
+	d.Set("task_name", siteMonitor.TaskName)
+	d.Set("task_type", siteMonitor.TaskType)
+	d.Set("task_state", siteMonitor.TaskState)
+	d.Set("interval", siteMonitor.Interval)
+	d.Set("options_json", siteMonitor.OptionsJson)
+	d.Set("create_time", siteMonitor.CreateTime)
+	d.Set("update_time", siteMonitor.UpdateTime)
+
+	return nil
+}
+
+func resourceAlicloudCmsSiteMonitorUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	request := cms.CreateModifySiteMonitorRequest()
+	request.TaskId = d.Id()
+	request.Address = d.Get("address").(string)
+	request.Interval = strconv.Itoa(d.Get("interval").(int))
+	request.OptionsJson = d.Get("options_json").(string)
+	request.TaskName = d.Get("task_name").(string)
+	alertIds := d.Get("alert_ids").([]interface{})
+	alertId := getAlertId(alertIds)
+	if alertId != "" {
+		request.AlertIds = alertId
+	}
+
+	_, err := client.WithCmsClient(func(cmsClient *cms.Client) (interface{}, error) {
+		return cmsClient.ModifySiteMonitor(request)
+	})
+	if err != nil {
+		return WrapError(err)
+	}
+
+	return resourceAlicloudCmsSiteMonitorRead(d, meta)
+}
+
+func resourceAlicloudCmsSiteMonitorDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	cmsService := CmsService{client}
+	request := cms.CreateDeleteSiteMonitorsRequest()
+
+	request.TaskIds = d.Id()
+	request.IsDeleteAlarms = "false"
+	taskName := d.Id()
+
+	return resource.Retry(3*time.Minute, func() *resource.RetryError {
+		_, err := client.WithCmsClient(func(cmsClient *cms.Client) (interface{}, error) {
+			return cmsClient.DeleteSiteMonitors(request)
+		})
+
+		if err != nil {
+			return resource.NonRetryableError(fmt.Errorf("Deleting site monitor got an error: %#v", err))
+		}
+
+		_, err = cmsService.DescribeSiteMonitor(taskName)
+		if err != nil {
+			if NotFoundError(err) {
+				return nil
+			}
+			return resource.NonRetryableError(fmt.Errorf("DescribeSiteMonitor got an error: %#v", err))
+		}
+
+		return resource.RetryableError(fmt.Errorf("Deleting site monitor got an error: %#v", err))
+
+	})
+}
+
+func getAlertId(alertIds []interface{}) string {
+	if alertIds != nil && len(alertIds) > 0 {
+		alertId := alertIds[0].(string)
```

Comment:
Why only use the first one? If so, I think the alert_ids should use alert_id.

Reply:
because also according to documentation at https://www.alibabacloud.com/help/doc-detail/115048.htm  ´alertIds´ is actually only a single value. There is no documented way to submit multiple ids (for example by delimiting them with a comma).

Reply:
done. multiple alert ids are ow seperated by comma as supported by API.